Objects/typeobject.c

```diff
@@ -3666,6 +3666,32 @@ PyType_FromMetaclass(PyTypeObject *metaclass, PyObject *module,
         goto finally;
     }
 
+    /* If this is an immutable type, check if all bases are also immutable,
+     * and (for now) fire a deprecation warning if not.
+     * (This isn't necessary for static types: those can't have heap bases,
+     * and only heap types can be mutable.)
+     */
+    if (spec->flags & Py_TPFLAGS_IMMUTABLETYPE) {
```

Comment:
```suggestion
    if (spec->flags & (Py_TPFLAGS_IMMUTABLETYPE | Py_TPFLAGS_HEAPTYPE)) {
```

Reply:
PyType_From* [only creates heap types](https://github.com/python/cpython/blob/07afe1b3f9c12b686b2c0bb4e3075000eea75bf7/Objects/typeobject.c#L3741).

Reply:
Doh, of course :)